tests/test_sourceSelector.py

```diff
@@ -0,0 +1,227 @@
+        self.config.unresolved.maximum = None
+        self.check((starGalaxy > minimum).tolist())
+
+
+class ReferenceSourceSelectorTaskTest(SourceSelectorTester, lsst.utils.tests.TestCase):
+    Task = ReferenceSourceSelectorTask
+
+    def setUp(self):
+        SourceSelectorTester.setUp(self)
+        self.config.magLimit.fluxField = "flux"
+        self.config.doMagLimit = True
+        self.config.doFlags = True
+
+    def testMagnitudeLimit(self):
+        tooBright = self.catalog.addNew()
+        tooBright.set("flux", 1.0e10)
+        tooBright.set("flux_flag", False)
+        good = self.catalog.addNew()
+        good.set("flux", 1000.0)
+        good.set("flux_flag", False)
+        bad = self.catalog.addNew()
+        bad.set("flux", good.get("flux"))
+        bad.set("flux_flag", True)
+        tooFaint = self.catalog.addNew()
+        tooFaint.set("flux", 1.0)
+        tooFaint.set("flux_flag", False)
+        # Note: magnitudes are backwards, so the minimum flux is the maximum magnitude
+        self.config.magLimit.minimum = lsst.afw.image.abMagFromFlux(1.0e6)
+        self.config.magLimit.maximum = lsst.afw.image.abMagFromFlux(10.0)
+        self.config.magLimit.fluxField = "flux"
+        self.check([False, True, False, False])
+
+        # Works with no minimum set?
+        self.config.magLimit.minimum = None
+        self.check([True, True, False, False])
+
+        # Works with no maximum set?
+        self.config.magLimit.maximum = None
+        self.check([True, True, False, True])
+
+
+    def testColorLimits(self):
+        num = 10
+        for _ in range(num):
+            self.catalog.addNew()
+        color = np.linspace(-0.5, 0.5, num, True)
+        flux = 1000.0
+        # Definition: color = mag(flux) - mag(otherFlux)
+        otherFlux = lsst.afw.image.fluxFromABMag(lsst.afw.image.abMagFromFlux(flux) - color)
+        self.catalog["flux"] = flux
+        self.catalog["other_flux"] = otherFlux
+        minimum, maximum = -0.1, 0.2
+        self.config.colorLimits = {"test": ColorLimit(primary="flux", secondary="other_flux",
+                                                      minimum=minimum, maximum=maximum)}
+        self.check(((color > minimum) & (color < maximum)).tolist())
+
+        # Works with no minimum set?
+        self.config.colorLimits["test"].minimum = None
+        self.check((color < maximum).tolist())
+
+        # Works with no maximum set?
+        self.config.colorLimits["test"].maximum = None
+        self.config.colorLimits["test"].minimum = minimum
+        self.check((color > minimum).tolist())
+
+        # Multiple limits
+        self.config.colorLimits = {"test": ColorLimit(primary="flux", secondary="other_flux", minimum=minimum),
+                                   "other": ColorLimit(primary="flux", secondary="other_flux",
+                                                       maximum=maximum)}
+        self.check(((color > minimum) & (color < maximum)).tolist())
```

Comment:
Do your minimum and maximum result in non-mutually-exclusive ranges here? I think so, but it might be worth checking.